java/statestore/src/main/java/sleeper/statestore/transactionlog/TransactionLogTransactionDeleter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.statestore.transactionlog;
+
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.core.statestore.transactionlog.TransactionLogStore;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+import static sleeper.configuration.properties.table.TableProperty.TRANSACTION_LOG_NUMBER_BEHIND_TO_DELETE;
+import static sleeper.configuration.properties.table.TableProperty.TRANSACTION_LOG_SNAPSHOT_MIN_AGE_MINUTES_TO_DELETE_TRANSACTIONS;
+
+/**
+ * Given a state store, finds transactions that are old enough to be deleted and deletes them.
```

Comment:
This is only relevant for a transaction log state store, so the start of this sentence should be "Given a transaction log state store".